blocks/store/reducer.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { filter, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { combineReducers } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Module Constants
+ */
+export const DEFAULT_CATEGORIES = [
+	{ slug: 'common', title: __( 'Common Blocks' ) },
+	{ slug: 'formatting', title: __( 'Formatting' ) },
+	{ slug: 'layout', title: __( 'Layout Elements' ) },
+	{ slug: 'widgets', title: __( 'Widgets' ) },
+	{ slug: 'embed', title: __( 'Embeds' ) },
+	{ slug: 'shared', title: __( 'Shared Blocks' ) },
+];
+
+/**
+ * Reducer managing the block types
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function blockTypes( state = { types: [] }, action ) {
```

Comment:
Seems like storing this as a deep object needlessly complicates things. Why not `blockTypes` as the object of `blockName: blockType`, and separate reducers for `defaultBlockName`, `fallbackBlockName` ? Or if we do want it to be nested, create via `combineReducers` ?

Reply:
So there are two questions here:

 - Why it's an array instead of an object: I wanted to keep the order of the registration similar to the order of `getBlockTypes`.

 - Why it's nested: I thought originally it was needed to be able to use part of the st To be able to reset the `defaultBlockName` and `fallbackBlockName` when we remove the block, but you're right it's not I'll update using `combineReducers`